geofence/geofence.go

```diff
@@ -0,0 +1,66 @@
+package echogeofence
+
+import (
+	"net"
+	"net/http"
+	"strings"
+
+	"github.com/circa10a/go-geofence"
+	"github.com/labstack/echo/v4"
+)
+
+// EchoGeofence holds a geofence config
+type EchoGeofence geofence.Config
+
+// config returns a geofence config from go-geofence
+func (e *EchoGeofence) config() *geofence.Config {
+	return &geofence.Config{
+		IPAddress: e.IPAddress,
+		Token:     e.Token,
+		Radius:    e.Radius,
+		CacheTTL:  e.CacheTTL,
+	}
+}
+
+// Middleware looks up the coordinates of a client to see if it's nearby
+func Middleware(echoGeofence *EchoGeofence) echo.MiddlewareFunc {
+	return func(next echo.HandlerFunc) echo.HandlerFunc {
+		return func(c echo.Context) error {
+			// Get client ip, strip port
+			remoteAddr := c.Request().RemoteAddr
```

Comment:
It would be better to use 	`echo.IPExtractor` to extract IP. `RemoteAddr` is not always actual IP of the client. See this https://echo.labstack.com/guide/ip-address/